### ISSUE TITLE
Talk.jekyll.com is down

### ISSUE THREAD

**Issue by austinpray** [jekyll/jekyll#3726]
https://talk.jekyllrb.com has been down for a couple hours. 

http://i.imgur.com/TMd4Ka3.jpg

**Comment by kleinfreund** [comment]
They probably know: [Parker on Twitter](https://twitter.com/parkr/status/601729772188606464)

> whaat? why do mx and cname records collide like this? didn't happen with mx and a records...

**Comment by envygeeks** [comment]
That comment from Parkr was actually a result of the changes we made so that I could keep talk.jekyllrb.com up without needing keys to the DNS.  We switched to a CNAME and apparently the DNS provider doesn't support that (not that many do even if it's not a violation of the RFC as far as I'm aware.)

The reason we are down was because there was a hitch in the migration (that was actually planned for Monday...) that forced us to do an emergency migration by hand, then the backups had hiccups (since resolved -- apparently duplicity created double chains -- easy fix) and now we are migrating to a new Docker setup (that does not use Discourse Docker but our own Discourse docker image) that is much more organized (for us at least... because it allows individual scaling, clustering and moving of Docker resources.) We should be back up within a few hours.  Sorry for the downtime.

**Comment by envygeeks** [comment]
TLDR version: I sacrificed a few hours of downtime today by force so I could finally get us into a state where I could prevent us from ever going down again to prevent that downtime later... because we were already down from the forced early migration.

**Comment by wjdp** [comment]
> We switched to a CNAME and apparently the DNS provider doesn't support that (not that many do even if it's not a violation of the RFC as far as I'm aware.)

Out of interest could you expand on that, CNAME for the talk subdomain or for the root?

**Comment by envygeeks** [comment]
By RFC an MX cannot point to a record if it is a CNAME.
Our root record is NOT a CNAME.
talk.jekyllrb.com is a CNAME.
the MX that talk.jekyllrb.com was going to point to was not a CNAME.

The problem we had is that talk.jekyllrb.com is a CNAME and we tried to create an MX for talk.jekyllrb.com but the DNS provider would have none of that (that is not a violation of the RFC the RFC violation would have been if we had pointed the MX to a CNAME not if the MX is the same as a CNAME pointing to an A record.)

**Comment by parkr** [comment]
Just migrated over to CloudFlare and waiting for everything to settle. The site works again and the emails should start working again in a day or two.

**Comment by envygeeks** [comment]
I'm closing this issue now, even though part of the Gravatar issue still exists that is set on the internal tracker for me to check on in the next few days and the bulk of this problem has since been solved.  Actually all but the gravatar issue has been solved.

**Comment by austinpray** [comment]
Good job, friends!

**Comment by wjdp** [comment]
Thanks for the explanation @envygeeks, interesting to hear about DNS issues with a bit of complexity
